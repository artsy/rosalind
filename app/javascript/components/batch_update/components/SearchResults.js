import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import ArtworkPreviewModal from './ArtworkPreviewModal'
import ArtworkSearchResult from './ArtworkSearchResult'
import Spinner from './Spinner'
import { Button } from '@artsy/palette'
import { Link } from './Links'

class SearchResults extends React.Component {
  maybeRenderSpinner() {
    const { isLoading } = this.props
    if (isLoading) {
      return <Spinner />
    } else {
      return null
    }
  }

  maybeRenderModal() {
    const {
      previewedArtwork,
      onPreviewArtwork,
      onPreviewPrevious,
      onPreviewNext,
    } = this.props
    if (previewedArtwork) {
      return (
        previewedArtwork && (
          <ArtworkPreviewModal
            artwork={previewedArtwork}
            onPreviewArtwork={onPreviewArtwork}
            onPreviewPrevious={onPreviewPrevious}
            onPreviewNext={onPreviewNext}
          />
        )
      )
    } else {
      return null
    }
  }

  maybeRenderControls() {
    const {
      artworks,
      totalHits,
      onSelectAllArtworks,
      onDeselectAllArtworks,
    } = this.props
    if (totalHits && artworks && artworks.length > 0) {
      return (
        <Controls
          displayed={artworks.length}
          total={totalHits}
          onSelectAllArtworks={onSelectAllArtworks}
          onDeselectAllArtworks={onDeselectAllArtworks}
        />
      )
    } else {
      return null
    }
  }

  maybeRenderMoreButton() {
    const { artworks, totalHits, onLoadMore } = this.props
    if (totalHits > artworks.length) {
      return <LoadMore onLoadMore={onLoadMore} />
    } else {
      return null
    }
  }

  render() {
    const {
      className,
      artworks,
      selectedArtworkIds,
      onPreviewArtwork,
      onToggleArtwork,
    } = this.props
    return (
      <div className={className}>
        {this.maybeRenderSpinner()}
        {this.maybeRenderModal()}
        {this.maybeRenderControls()}
        <ArtworkResultList
          artworks={artworks}
          selectedArtworkIds={selectedArtworkIds}
          onPreviewArtwork={onPreviewArtwork}
          onToggleArtwork={onToggleArtwork}
        />
        {this.maybeRenderControls()}
        {this.maybeRenderMoreButton()}
      </div>
    )
  }
}

SearchResults.propTypes = {
  artworks: PropTypes.array.isRequired,
  isLoading: PropTypes.bool,
  onDeselectAllArtworks: PropTypes.func,
  onLoadMore: PropTypes.func,
  onPreviewArtwork: PropTypes.func,
  onPreviewNext: PropTypes.func,
  onPreviewPrevious: PropTypes.func,
  onSelectAllArtworks: PropTypes.func,
  onToggleArtwork: PropTypes.func,
  previewedArtwork: PropTypes.object,
  selectedArtworkIds: PropTypes.arrayOf(PropTypes.string),
  totalHits: PropTypes.number,
}

const Controls = ({
  displayed,
  total,
  onSelectAllArtworks,
  onDeselectAllArtworks,
}) => (
  <div>
    <div className="counts">
      Displaying {displayed.toLocaleString()} of {total.toLocaleString()}{' '}
      matching artworks
    </div>
    <div className="select">
      Select:
      <Link
        href="#"
        onClick={e => {
          e.preventDefault()
          onSelectAllArtworks()
        }}
      >
        all
      </Link>
      /
      <Link
        href="#"
        onClick={e => {
          e.preventDefault()
          onDeselectAllArtworks()
        }}
      >
        none
      </Link>
    </div>
  </div>
)

const ArtworkResultList = ({
  artworks,
  selectedArtworkIds,
  onPreviewArtwork,
  onToggleArtwork,
}) => {
  return (
    <div className="results">
      {artworks.map(artwork => (
        <ArtworkSearchResult
          key={artwork.id}
          artwork={artwork}
          onPreviewArtwork={onPreviewArtwork}
          onToggleArtwork={onToggleArtwork}
          selected={selectedArtworkIds.indexOf(artwork.id) > -1}
        />
      ))}
    </div>
  )
}

const LoadMore = ({ onLoadMore }) => (
  <Button onClick={onLoadMore}>Load more</Button>
)

/* default styled component */

const StyledSearchResults = styled(SearchResults)`
  display: flex;
  flex-direction: column;

  .counts {
    display: inline-block;
    width: 75%;
    padding: 0.75em;
  }

  .select {
    display: inline-block;
    width: 25%;
    padding: 0.75em;
    text-align: right;

    a {
      margin: 0 0.25em;
    }
  }

  .results {
    display: flex;
    flex-flow: row wrap;
  }
`

export default StyledSearchResults
