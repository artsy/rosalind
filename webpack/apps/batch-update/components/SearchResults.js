import React from 'react'
import missingImage from 'file-loader!./missing_image.png'
import ArtworkPreviewModal from './ArtworkPreviewModal'
import Spinner from './Spinner'
import { Button } from './Buttons'

class SearchResults extends React.Component {
  maybeRenderSpinner () {
    const { isLoading } = this.props
    if (isLoading) {
      return <Spinner />
    } else {
      return null
    }
  }

  maybeRenderModal () {
    const { previewedArtwork, onPreviewArtwork, onPreviewPrevious, onPreviewNext } = this.props
    if (previewedArtwork) {
      return previewedArtwork && <ArtworkPreviewModal
        artwork={previewedArtwork}
        onPreviewArtwork={onPreviewArtwork}
        onPreviewPrevious={onPreviewPrevious}
        onPreviewNext={onPreviewNext}
      />
    } else {
      return null
    }
  }

  maybeRenderCounts () {
    const { artworks, totalHits } = this.props
    if (totalHits && artworks && artworks.length > 0) {
      return <Counts displayed={artworks.length} total={totalHits} />
    } else {
      return null
    }
  }

  maybeRenderMoreButton () {
    const { artworks, totalHits, onLoadMore } = this.props
    if (totalHits > artworks.length) {
      return <LoadMore onLoadMore={onLoadMore} />
    } else {
      return null
    }
  }

  render () {
    const { className, artworks, onPreviewArtwork } = this.props
    return (
      <div className={className}>
        {this.maybeRenderSpinner()}
        {this.maybeRenderModal()}
        {this.maybeRenderCounts()}
        <ArtworkResultList artworks={artworks} onPreviewArtwork={onPreviewArtwork} />
        {this.maybeRenderMoreButton()}
      </div>
    )
  }
}

SearchResults.propTypes = {
  artworks: React.PropTypes.array.isRequired,
  isLoading: React.PropTypes.bool,
  previewedArtwork: React.PropTypes.object,
  onPreviewArtwork: React.PropTypes.func,
  onPreviewPrevious: React.PropTypes.func,
  onPreviewNext: React.PropTypes.func
}

const Counts = ({displayed, total}) => (
  <div className='counts'>
    Displaying {displayed.toLocaleString()} of {total.toLocaleString()} matching artworks
  </div>
)

const ArtworkResultList = ({artworks, onPreviewArtwork}) => {
  return (
    <div className='results'>
      {artworks.map(artwork => <ArtworkResult key={artwork.id} artwork={artwork} onPreviewArtwork={onPreviewArtwork} />)}
    </div>
  )
}

const ArtworkResult = ({artwork, onPreviewArtwork}) => {
  const { name, image_url: imageUrl } = artwork
  return (
    <div className='result' onClick={() => { onPreviewArtwork(artwork) }}>
      <img src={imageUrl || missingImage} alt={name} />
      <figcaption>{name}</figcaption>
    </div>
  )
}

const LoadMore = ({onLoadMore}) => (
  <Button primary fullWidth onClick={onLoadMore}>
    Load more
  </Button>
)

/* default styled component */

import styled from 'styled-components'

const StyledSearchResults = styled(SearchResults)`
  display: flex;
  flex-direction: column;

  .counts {
    padding: 0.75em;
  }

  .results {
    display: flex;
    flex-flow: row wrap;

    .result {
      min-width: 115px;
      max-width: calc(20% - 1.5em);
      margin: 0.75em;
      flex: 1 1 calc(20% - 1.5em);

      img {
        width: 100%;
      }

      figcaption {
        position: relative;
        font-size: 80%;
        max-height: 3.75em;
        line-height: 1.25em;
        overflow: hidden;
      }
    }
  }
`

export default StyledSearchResults
