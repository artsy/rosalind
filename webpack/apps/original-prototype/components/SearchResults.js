import React from 'react'
import missingImage from 'file-loader!./missing_image.png'

function SearchResults (props) {
  const { artworks, previewedArtwork, onPreviewArtwork, onPreviewPrevious, onPreviewNext } = props
  return (
    <div className='SearchResults'>
      {
        previewedArtwork ? <ArtworkPreview
          artwork={previewedArtwork}
          onPreviewArtwork={onPreviewArtwork}
          onPreviewPrevious={onPreviewPrevious}
          onPreviewNext={onPreviewNext}
          /> : null
      }
      {
        artworks.map((artwork) =>
          <Artwork key={artwork.id}
            artwork={artwork}
            onPreviewArtwork={onPreviewArtwork}
            />
        )
      }
    </div>
  )
}

class Artwork extends React.Component {
  constructor (props) {
    super(props)
    this.handleClick = this.handleClick.bind(this)
  }

  handleClick () {
    this.props.onPreviewArtwork(this.props.artwork)
  }

  render () {
    const { name, image_url: imageUrl } = this.props.artwork
    return (
      <div className='tmp-artwork' onClick={this.handleClick}>
        <img src={imageUrl || missingImage} alt={name} />
        <figcaption>{name}</figcaption>
      </div>
    )
  }
}

class ArtworkPreview extends React.Component {
  constructor (props) {
    super(props)
    this.dismiss = this.dismiss.bind(this)
    this.handleKeyUp = this.handleKeyUp.bind(this)
  }

  dismiss () {
    this.props.onPreviewArtwork(null)
  }

  prev () {
    this.props.onPreviewPrevious()
  }

  next () {
    this.props.onPreviewNext()
  }

  handleKeyUp (e) {
    e.keyCode === 27 && this.dismiss()
    e.keyCode === 37 && this.prev()
    e.keyCode === 39 && this.next()
  }

  componentDidMount () {
    window.addEventListener('keyup', this.handleKeyUp)
  }

  componentWillUnmount () {
    window.removeEventListener('keyup', this.handleKeyUp)
  }

  render () {
    const { id, artist_id: artistId, partner_id: partnerId, name, image_url: imageUrl, published, genomed, deleted } = this.props.artwork
    return (
      <div className='ArtworkPreview-Overlay' onClick={this.dismiss}>
        <div className='ArtworkPreview' ref={(el) => { this.modal = el }}>
          <div className='ArtworkPreview-Image'>
            <img src={imageUrl || missingImage} alt={name} />
          </div>
          <div className='ArtworkPreview-Details'>
            <p style={{fontWeight: 'bold'}}>{name}</p>
            <p style={{fontSize: '0.9em', lineHeight: '125%', margin: '1em 0'}}>
              Deleted: {deleted.toString()} <br />
              Published: {published.toString()} <br />
              Genomed: {genomed.toString()}
            </p>
            <p>
              View artwork in:
              <a style={{margin: '0 0.25em'}} href={`https://helix.artsy.net/genome/artworks?artwork_ids=${id}`}>Helix</a>|
              <a style={{margin: '0 0.25em'}} href={`https://cms.artsy.net/artworks/${id}/edit?current_partner_id=${partnerId}`}>CMS</a>|
              <a style={{margin: '0 0.25em'}} href={`https://www.artsy.net/artwork/${id}`}>Artsy.net</a>
            </p>
            <p>
              View artist in:
              <a style={{margin: '0 0.25em'}} href={`https://helix.artsy.net/genome/artist?search[genome_artist_id]=${artistId}`}>Helix</a>
            </p>
          </div>
        </div>
      </div>
    )
  }
}

export default SearchResults
