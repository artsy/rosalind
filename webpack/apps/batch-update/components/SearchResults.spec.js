import React from 'react'
import renderer from 'react-test-renderer'
import SearchResults from './SearchResults'

let soup, shark, artworks

beforeEach(() => {
  soup = {
    id: 'can',
    name: 'Soup Can',
    image_url: 'can.jpg',
    deleted: false,
    published: true,
    genomed: true
  }
  shark = {
    id: 'shark',
    name: 'Shark',
    image_url: 'shark.jpg',
    deleted: false,
    published: true,
    genomed: true
  }
  artworks = [soup, shark]
})

it('renders a collection of artworks', () => {
  const rendered = renderer.create(<SearchResults artworks={artworks} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders a modal when previewing', () => {
  const rendered = renderer.create(<SearchResults previewedArtwork={soup} artworks={artworks} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('does not render a modal when not previewing', () => {
  const rendered = renderer.create(<SearchResults previewedArtwork={null} artworks={artworks} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders a spinner while fetching artworks', () => {
  const rendered = renderer.create(<SearchResults isLoading previewedArtwork={null} artworks={artworks} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})
