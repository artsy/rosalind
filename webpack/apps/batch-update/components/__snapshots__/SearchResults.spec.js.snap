// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render a modal when not previewing 1`] = `
.c1 {
  min-width: 115px;
  max-width: calc(20% - 1.5em);
  margin: 0.75em;
  -webkit-flex: 1 1 calc(20% - 1.5em);
  -ms-flex: 1 1 calc(20% - 1.5em);
  flex: 1 1 calc(20% - 1.5em);
}

.c1 img {
  width: 100%;
}

.c1 figcaption {
  position: relative;
  font-size: 80%;
  max-height: 3.75em;
  line-height: 1.25em;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .counts {
  display: inline-block;
  width: 75%;
  padding: 0.75em;
}

.c0 .select {
  display: inline-block;
  width: 25%;
  padding: 0.75em;
  text-align: right;
}

.c0 .select a {
  margin: 0 0.25em;
}

.c0 .results {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

<div
  className="c0"
>
  <div
    className="results"
  >
    <div
      className="c1"
      onClick={[Function]}
    >
      <img
        alt="Soup Can"
        src="can.jpg"
      />
      <figcaption>
        Soup Can
      </figcaption>
    </div>
    <div
      className="c1"
      onClick={[Function]}
    >
      <img
        alt="Shark"
        src="shark.jpg"
      />
      <figcaption>
        Shark
      </figcaption>
    </div>
  </div>
</div>
`;

exports[`renders a collection of artworks 1`] = `
.c1 {
  min-width: 115px;
  max-width: calc(20% - 1.5em);
  margin: 0.75em;
  -webkit-flex: 1 1 calc(20% - 1.5em);
  -ms-flex: 1 1 calc(20% - 1.5em);
  flex: 1 1 calc(20% - 1.5em);
}

.c1 img {
  width: 100%;
}

.c1 figcaption {
  position: relative;
  font-size: 80%;
  max-height: 3.75em;
  line-height: 1.25em;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .counts {
  display: inline-block;
  width: 75%;
  padding: 0.75em;
}

.c0 .select {
  display: inline-block;
  width: 25%;
  padding: 0.75em;
  text-align: right;
}

.c0 .select a {
  margin: 0 0.25em;
}

.c0 .results {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

<div
  className="c0"
>
  <div
    className="results"
  >
    <div
      className="c1"
      onClick={[Function]}
    >
      <img
        alt="Soup Can"
        src="can.jpg"
      />
      <figcaption>
        Soup Can
      </figcaption>
    </div>
    <div
      className="c1"
      onClick={[Function]}
    >
      <img
        alt="Shark"
        src="shark.jpg"
      />
      <figcaption>
        Shark
      </figcaption>
    </div>
  </div>
</div>
`;

exports[`renders a modal when previewing 1`] = `
.c1 {
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background: hsla(0,0%,0%,0.5);
  z-index: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  box-sizing: border-box;
  padding: 2em;
  max-height: 90%;
  z-index: 2;
  background: white;
  box-shadow: 0 0 20px hsla(0,0%,0%,0.5);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c2 .image {
  -webkit-flex: 0 1 40%;
  -ms-flex: 0 1 40%;
  flex: 0 1 40%;
}

.c2 .details {
  max-width: 20em;
  margin-left: 1em;
  -webkit-flex: 0 1 60%;
  -ms-flex: 0 1 60%;
  flex: 0 1 60%;
}

.c2 .name {
  font-weight: bold;
}

.c2 .status {
  font-size: 0.9em;
  line-height: 125%;
  margin: 1em 0;
}

.c2 .links a {
  margin: 0 0.25em;
}

.c3 {
  min-width: 115px;
  max-width: calc(20% - 1.5em);
  margin: 0.75em;
  -webkit-flex: 1 1 calc(20% - 1.5em);
  -ms-flex: 1 1 calc(20% - 1.5em);
  flex: 1 1 calc(20% - 1.5em);
}

.c3 img {
  width: 100%;
}

.c3 figcaption {
  position: relative;
  font-size: 80%;
  max-height: 3.75em;
  line-height: 1.25em;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .counts {
  display: inline-block;
  width: 75%;
  padding: 0.75em;
}

.c0 .select {
  display: inline-block;
  width: 25%;
  padding: 0.75em;
  text-align: right;
}

.c0 .select a {
  margin: 0 0.25em;
}

.c0 .results {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

<div
  className="c0"
>
  <div
    className="c1"
    onClick={[Function]}
    z={1}
  >
    <div
      className="c2"
      onClickCapture={[Function]}
    >
      <div
        className="image"
      >
        <img
          alt="Soup Can"
          src="can.jpg"
        />
      </div>
      <div
        className="details"
      >
        <p
          className="name"
        >
          Soup Can
        </p>
        <p
          className="status"
        >
          Deleted: 
          false
           
          <br />
          Published: 
          true
           
          <br />
          Genomed: 
          true
        </p>
        <p
          className="links"
        >
          View artwork in:
          <a
            href="https://helix.artsy.net/genome/artworks?artwork_ids=can"
            target="_blank"
          >
            Helix
          </a>
          |
          <a
            href="https://cms.artsy.net/artworks/can/edit?current_partner_id=undefined"
            target="_blank"
          >
            CMS
          </a>
          |
          <a
            href="https://www.artsy.net/artwork/can"
            target="_blank"
          >
            Artsy.net
          </a>
        </p>
        <p
          className="links"
        >
          View artist in:
          <a
            href="https://helix.artsy.net/genome/artist?search[genome_artist_id]=undefined"
            target="_blank"
          >
            Helix
          </a>
        </p>
      </div>
    </div>
  </div>
  <div
    className="results"
  >
    <div
      className="c3"
      onClick={[Function]}
    >
      <img
        alt="Soup Can"
        src="can.jpg"
      />
      <figcaption>
        Soup Can
      </figcaption>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <img
        alt="Shark"
        src="shark.jpg"
      />
      <figcaption>
        Shark
      </figcaption>
    </div>
  </div>
</div>
`;

exports[`renders a spinner while fetching artworks 1`] = `
.c3 {
  min-width: 115px;
  max-width: calc(20% - 1.5em);
  margin: 0.75em;
  -webkit-flex: 1 1 calc(20% - 1.5em);
  -ms-flex: 1 1 calc(20% - 1.5em);
  flex: 1 1 calc(20% - 1.5em);
}

.c3 img {
  width: 100%;
}

.c3 figcaption {
  position: relative;
  font-size: 80%;
  max-height: 3.75em;
  line-height: 1.25em;
  overflow: hidden;
}

.c1 {
  position: absolute;
  left: 0;
  top: 0;
  right: 0;
  height: calc(100vh - 50px);
  background: hsla(0,0%,100%,0.90);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  z-index: 1;
}

.c2 {
  width: 12%;
  height: 1%;
  background: black;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  z-index: 2;
  -webkit-animation: MQNGw 2s linear infinite;
  animation: MQNGw 2s linear infinite;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .counts {
  display: inline-block;
  width: 75%;
  padding: 0.75em;
}

.c0 .select {
  display: inline-block;
  width: 25%;
  padding: 0.75em;
  text-align: right;
}

.c0 .select a {
  margin: 0 0.25em;
}

.c0 .results {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
      color="black"
    />
  </div>
  <div
    className="results"
  >
    <div
      className="c3"
      onClick={[Function]}
    >
      <img
        alt="Soup Can"
        src="can.jpg"
      />
      <figcaption>
        Soup Can
      </figcaption>
    </div>
    <div
      className="c3"
      onClick={[Function]}
    >
      <img
        alt="Shark"
        src="shark.jpg"
      />
      <figcaption>
        Shark
      </figcaption>
    </div>
  </div>
</div>
`;
